### PR TITLE
Add Script to Monitor CPU Temperature Using AMD Thermal Sensors on OPNsense

### PR DESCRIPTION
**Pull request by fnateghi**
### Description:
This pull request introduces a new Bash script designed to monitor CPU temperatures using AMD thermal sensors on OPNsense systems. It reads temperature data through sysctl and issues alerts based on set thresholds.

### Changes:

New Script: Implements a script to fetch temperatures from dev.amdtemp.0.core0.sensor0.
Thresholds: Configures warnings at 60°C and critical alerts at 80°C.
Output: Provides detailed outputs including temperature readings, status codes, and contextual alert messages.
### Prerequisites:

Configuration Requirement: Ensure AMD thermal sensors are enabled under System Settings > Miscellaneous > Thermal Sensors > Hardware in OPNsense.
### Testing Conducted:

Environment: Tested on an OPNsense system with configured AMD thermal sensors.
Results: The script successfully identified temperature fluctuations and triggered alerts at defined thresholds.

### CODE REVIEW

**Review by zerwes — CHANGES_REQUESTED** [review]

**Comment by fnateghi** [comment]
Ive added the requested changes into the script and README file.

File: files/cputemp.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+        # Determine status based on temperature thresholds
+        if (( $(echo "$temp >= $temp_crit" | bc -l) )); then
+            ECODE=2
+            TXT="CPU Temp critical"
+        elif (( $(echo "$temp >= $temp_warn" | bc -l) )); then
+            ECODE=1
+            TXT="CPU Temp warning"
+        fi 
+        echo "$ECODE CPUTEMP - temperature is $temp°C - $TXT"
```
Comment (zerwes) [inline files/cputemp.sh:27]: the checks need to have different names, so checkmk can distinguish them
```diff
-        echo "$ECODE CPUTEMP - temperature is $temp°C - $TXT"
+        echo "$ECODE CPUTEMP-$sensor_id - temperature is $temp°C - $TXT"
```

File: files/cputemp.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#!/usr/local/bin/bash
+
+declare temp_warn=60
+declare temp_crit=80
+
+# Function to display temperatures for all found sensors of a specific type
+display_temps() {
+    local sensor_pattern=$1
+    local description=$2
```
Comment (zerwes) [inline files/cputemp.sh:9]: description is not used, so no need to pass the arg arround
```diff
@@ -6,7 +6,6 @@
 # Function to display temperatures for all found sensors of a specific type
 display_temps() {
     local sensor_pattern=$1
-    local description=$2
 
     # Collect all relevant sensor readings
     sysctl -a | grep -E "$sensor_pattern" | while read sensor_data; do
@@ -30,13 +29,13 @@
 # Main script execution to check different sensor types
 # Check AMD temperature sensors
 if sysctl -a | grep -q 'dev.amdtemp.0.core0.sensor0'; then
-    display_temps 'dev.amdtemp.[0-9]+.core[0-1].sensor[0-1]' "AMD CPU Sensor"
+    display_temps 'dev.amdtemp.[0-9]+.core[0-1].sensor[0-1]'
 # Check Intel CPU temperature sensors
 elif sysctl -a | grep -q 'dev.cpu.0.temperature'; then
-    display_temps 'dev.cpu.[0-9]+.temperature' "Intel CPU Sensor"
+    display_temps 'dev.cpu.[0-9]+.temperature'
 # Check PCH temperature sensors, if any
 elif sysctl -a | grep -q 'dev.pchtherm.0.temperature'; then
-    display_temps 'dev.pchtherm.[0-9]+.temperature' "PCH Sensor"
+    display_temps 'dev.pchtherm.[0-9]+.temperature'
 else
```

**Review by zerwes — CHANGES_REQUESTED** [review]
Hello @fnateghi 
Thank you for the fast and very good changes in order to make the check much more generic.

I just suggest 2 minimal additional changes.

Thank you for your contribution.

**Comment by fnateghi** [comment]
I edited the required changes. Now should be fine.

**Comment by fnateghi** [comment]
@zerwes, I'm glad to contribute to this issue. Even though it's a small change, I believe some people will find it useful.

**Review by zerwes — APPROVED** [review]
Thank you for your fix

**Pull request merged**